pynest/nest/lib/hl_api_connection_helpers.py

```diff
@@ -236,3 +248,71 @@ def fixdict(d):
     projections = fixdict(projections)
     sps(projections)
     sr('ConnectLayers')
+
+
+def _process_input_nodes(pre, post, conn_spec):
+    """
+    Check the properties of `pre` and `post` nodes:
+
+    * If `conn_spec` is 'one_to_one', no uniqueness check is performed; the
+      "regular" one-to-one connect is used if both inputs are NodeCollection,
+      "connect_arrays" is used otherwise.
+    * If both `pre` and `post` are NodeCollections or can be converted to
+      NodeCollections (i.e. contain unique IDs), then proceed to "regular"
+      connect (potentially after conversion to NodeCollection).
+    * If both `pre` and `post` are arrays and contain non-unique items, then
+      we proceed to "connect_arrays".
+    * If at least one of them has non-unique items and they have different
+      sizes, then raise an error.
+    """
+    use_connect_arrays = False
+
+    # check for 'one_to_one' conn_spec
+    one_to_one_cspec = (conn_spec == 'one_to_one')
+
+    if isinstance(conn_spec, dict):
+        one_to_one_cspec = (conn_spec.get('rule', 'all_to_all') == 'one_to_one')
```

Comment:
Setting `one_to_one_cspec` first and then conditionally changing it right afterwards seems confusing. How about
```suggestion
    one_to_one_cspec = ( ( conn_spec if not isinstance(conn_spec, dict) 
                                            else conn_spec.get('rule', 'all_to_all') )
                                          == 'one_to_one' ) 
```
I intentionally put the `if not` in so the plain `conn_spec` comes first.

Reply:
Because I found the original structure more readable but I don't care much so I'll change it